### PR TITLE
feature(ui-ux): transaction card component 2.0

### PR DESCRIPTION
**Pull request by chloezxyy**
#### What kind of PR is this?:

/kind feature

#### Which issue(s) does this PR fixes?:

Fixes #

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [ ] No console errors on web
- [ ] Tested on Light mode and Dark mode*
- [ ] Your UI implementation visually matched the rendered design*
- [ ] Unit tests*
- [ ] Added e2e tests*
- [ ] Added translations*

### CODE REVIEW

File: mobile-app/app/screens/AppNavigator/screens/Dex/components/TransactionCard.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  max = 'MAX'
+}
+
+export function TransactionCard (props: TransactionCardProps): JSX.Element {
+  const Icon = getNativeIcon(props.symbol)
+  return (
+    <ThemedViewV2
+      light={tailwind('bg-mono-light-v2-00')}
+      dark={tailwind('bg-mono-dark-v2-00')}
+      style={tailwind('rounded-lg-v2 px-4 py-4')}
```
Comment (pierregee) [inline mobile-app/app/screens/AppNavigator/screens/Dex/components/TransactionCard.tsx]:
```suggestion
      style={tailwind('rounded-lg-v2 p-4')}
```

File: mobile-app/app/components/TransactionCard.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+import { translate } from '@translations'
+import { getNativeIcon } from '@components/icons/assets'
+import { StyleProp, ViewStyle } from 'react-native'
+
+type EditingAmount = 'primary' | 'secondary'
+
+interface TransactionCardProps {
+  symbol: string
+  balance: BigNumber
+  current: string
```
Comment (pierregee) [inline mobile-app/app/components/TransactionCard.tsx:14]:
```suggestion
  value: string
```

File: mobile-app/app/components/TransactionCard.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+import { WalletTextInputV2 } from '@components/WalletTextInputV2'
+import { translate } from '@translations'
+import { getNativeIcon } from '@components/icons/assets'
+import { StyleProp, ViewStyle } from 'react-native'
+
+type EditingAmount = 'primary' | 'secondary'
+
+interface TransactionCardProps {
+  symbol: string
+  balance: BigNumber
```
Comment (pierregee) [inline mobile-app/app/components/TransactionCard.tsx:13]:
```suggestion
  maxValue: BigNumber
```
just to be more generic with the naming

**Review by pierregee — CHANGES_REQUESTED** [review]

**Review by pierregee — DISMISSED** [review]

**Review by pierregee — DISMISSED** [review]

**Comment by thedoublejay** [comment]
@pierregee @chloezxyy let's merge this after the release

File: mobile-app/app/components/TransactionCard.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        'border-0.5 border-red-v2': status === 'error'
+      })}
+      style={tailwind('rounded-lg-v2', {
+        'border-0.5 border-red-v2': status === 'error'
+      })}
+    >
+      <ThemedViewV2
+        light={tailwind('bg-mono-light-v2-00')}
+        dark={tailwind('bg-mono-dark-v2-00')}
+        style={tailwind('pl-5 pr-5 pt-2 mr-px rounded-t-lg-v2')}
```
Comment (pierregee) [inline mobile-app/app/components/TransactionCard.tsx:37]:
```suggestion
        style={tailwind('rounded-t-lg-v2')}
```
to make way for other designs, let's move the padding and margins in `{children}` e.g send

File: mobile-app/app/components/TransactionCard.tsx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+      >
+        {children}
+      </ThemedViewV2>
+      <ThemedViewV2
+        light={tailwind('border-mono-light-v2-300')}
+        dark={tailwind('border-mono-dark-v2-300')}
+        style={tailwind('flex flex-row justify-around items-center pt-2 pb-4 border-t-0.5')}
```
Comment (pierregee) [inline mobile-app/app/components/TransactionCard.tsx:44]:
```suggestion
        style={tailwind('flex flex-row justify-around items-center pb-4')}
```
send has no border, can we move this on `{children}` bottom border too?

**Pull request closed without merging**